src/NuGet.Clients/NuGet.VisualStudio.Implementation/PreinstalledPackageInstaller.cs

```diff
@@ -160,18 +161,24 @@ internal string GetRegistryRepositoryPath(string keyName, IEnumerable<IRegistryK
             IVsPackageInstaller packageInstaller,
             EnvDTE.Project project,
             PreinstalledPackageConfiguration configuration,
+            bool preferPackageReferenceFormat,
             Action<string> warningHandler,
             Action<string> errorHandler)
         {
             ThreadHelper.ThrowIfNotOnUIThread();
 
-            string repositoryPath = configuration.RepositoryPath;
+            var repositoryPath = configuration.RepositoryPath;
             var repositorySource = new Configuration.PackageSource(repositoryPath);
             var failedPackageErrors = new List<string>();
 
             // find the project
             var defaultProjectContext = new VSAPIProjectContext();
+
             var nuGetProject = await _solutionManager.GetOrCreateProjectAsync(project, defaultProjectContext);
+            if (preferPackageReferenceFormat && await NuGetProjectUpgradeUtility.IsNuGetProjectUpgradeableAsync(nuGetProject, project, false))
```

Comment:
What is need for this boolean? when would we install package through these api in existing project?

Reply:
What do you mean by an existing project?
I'm making that call to not duplicate the logic of checking whether a project GUID is PR compatible.